src/Orleans.Runtime/GrainDirectory/GrainLocator.cs

```diff
@@ -160,26 +170,27 @@ public void Participate(ISiloLifecycle lifecycle)
         {
             Task onStart(CancellationToken ct)
             {
-                ListenToClusterChange().Ignore();
+                this.listenToClusterChangeTask = ListenToClusterChange();
                 return Task.CompletedTask;
             };
-            Task onStop(CancellationToken ct)
+            async Task onStop(CancellationToken ct)
             {
                 this.shutdownToken.Cancel();
-                return Task.CompletedTask;
+                await listenToClusterChangeTask;
```

Comment:
Check that it's not null first?

May also want to skip awaiting if `ct.IsCancellationRequested` - could this ever timeout?

BTW, I think all of our other local functions use upper case names.

Reply:
It could timeout yes, since we do a `await this.grainDirectory.UnregisterSilos(deadSilos)` in `ListenToClusterChange`. I will propagate the `shutdownToken` where necessary, I think that will be enough